index.js

```diff
@@ -208,43 +242,20 @@ async function main() {
     }
 }
 
-async function sleep(ms) {
-    return new Promise((resolve) => setTimeout(resolve, ms));
-}
-
-async function newCompletion({ messages }) {
-    let attempts = 0;
-    const maxAttempts = 99;
-
-    while (attempts < maxAttempts) {
-        try {
-            const output = await openai.chat.completions.create({
-                messages,
-                model: "gpt-4o",
-                max_tokens: 1000,
-                temperature: 0.0,
-                n: 1,
-                seed: 0,
-            });
+async function newCompletion({ messages, schema }) {
+    const { object } = await generateObject({
+        model: modelConfig,
+        messages,
+        temperature: 0.0,
+        maxRetries: 5,
+        maxTokens: 1000,
+        seed: 0,
+        schema,
+    });
 
-            logger.info(output.choices[0].message.content);
```

Comment:
Wouldn't we still need some rate-limiting retry mechanism? Or does vercels sdk have this baked in?

Reply:
Yeah, I went this way because I saw it baked in with a expoential backoff apparently:
https://sdk.vercel.ai/docs/ai-sdk-core/settings#maxretries

Want to give it a try and see if we need to bump the maxRetires above the default of 2?